### PR TITLE
Quickbooks custom Provider

### PR DESCRIPTION
**Pull request by tjcrowley**
This is a custom provider I wrote to do oauth2 auth with Intuit Quickbooks and Payments APIs. 

It requires that you set a QBO_TEST variable in your settings file. If it's true, it uses the sandbox URL. If not, it assume production and tries that URL. In my system this file is at config.settings.base, probably need help making that more generic. But it works.

### CODE REVIEW

**Comment by pennersr** [comment]
You can have a look at the base provider class. It features a method `get_settings`. This basically allows you to organize the provider settings, like so:

```
SOCIALACCOUNT_PROVIDERS = { 'quickbooks': { 'SANDBOX': True } }
```
Can you make use of that? Also, please add the snippet above in the `providers.rst` so that others know how to enable the sandbox mode.

**Comment by tjcrowley** [comment]
Thanks, I will push a fix later today!

File: allauth/socialaccount/providers/quickbooks/provider.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        first_name = self.account.extra_data.get('givenName', None)
+        last_name = self.account.extra_data.get('familyName', None)
+        if first_name and last_name:
+            name = first_name + ' ' + last_name
+        return name
+
+class QuickbooksOAuth2Provider(OAuth2Provider):
+    id = 'quickbooks'
+    # Name is displayed to ordinary users -- don't include protocol
+    name = 'Quickbooks'
```
Comment (pennersr) [inline allauth/socialaccount/providers/quickbooks/provider.py]: Sorry for nitpicking, can you name this one "QuickBooks" as that seems to be the official style. Please rename the classes as well (QuickbooksAccount -> QuickBooksAccount). Thanks.

File: allauth/socialaccount/providers/quickbooks/provider.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+from allauth.socialaccount import app_settings
```
Comment (pennersr) [inline allauth/socialaccount/providers/quickbooks/provider.py]: Can you add this provider to `test_settings.py`, `overview.rst` (list of supported providers) and `installation.rst` ?

File: allauth/socialaccount/providers/quickbooks/views.py [hunk 3]
```diff
@@ -32,7 +32,9 @@ def complete_login(self, request, app, token, **kwargs):
     def get_user_info(self, token):
         auth_header = 'Bearer ' + token.token
         headers = {'Accept': 'application/json', 'Authorization': auth_header, 'accept': 'application/json'}
-        if app_settings.PROVIDERS.quickbooks.SANDBOX:
+        QBO_sandbox = getattr(app_settings, 'SOCIALACCOUNT_PROVIDERS', {}).get(
```
Comment (pennersr) [inline allauth/socialaccount/providers/quickbooks/views.py]: You can replace this with self.get_provider().get_settings().get('SANDBOX', False)

Comment (tjcrowley) [reply allauth/socialaccount/providers/quickbooks/views.py]: Just figured this out from looking at other providers, thank you!

File: tox.ini [hunk 4]
```diff
@@ -32,6 +32,7 @@ commands =
 
 [testenv:flake8]
 skip_install = True
+max-line-length=90
```
Comment (pennersr) [inline tox.ini]: I am actually quite fond of pep8 and its line length limitations. Normally there is always a way to reduce the line length, and should you really have a case where readability is in danger, add a `# noqa` at the end of the line.

Comment (tjcrowley) [reply tox.ini]: Slowly figuring this out, I've never used Travis before. Thanks for the guidance.

**Pull request merged**